Tests/Runner/AuthorizationTest.php

```diff
@@ -47,13 +47,76 @@ public function testOauthDecrypt()
         $config = ['oauth_api' => ['id' => 'whatever']];
 
         /** @var Credentials $oauthClientStub */
-        $auth = new Authorization($oauthClientStub, $encryptorFactory->getEncryptor(), 'keboola.docker-demo', false);
+        $auth = new Authorization($oauthClientStub, $oauthClientStub, $encryptorFactory->getEncryptor(), 'keboola.docker-demo', false);
         $this->assertEquals(
             $credentials,
             $auth->getAuthorization($config)['oauth_api']['credentials']
         );
     }
 
+    public function testOauthDecryptVersions()
+    {
+        $encryptorFactory = new ObjectEncryptorFactory(
+            'alias/dummy-key',
+            'us-east-1',
+            hash('sha256', uniqid()),
+            hash('sha256', uniqid())
+        );
+        $encryptorFactory->setComponentId('keboola.docker-demo');
+
+        $oauthClientStub2 = self::getMockBuilder(Credentials::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $credentials2 = [
+            'id' => 'test-credentials-45',
+            '#data' => '{"access_token":"abcd","token_type":"bearer","uid":"efgh"}',
+            'oauthVersion' => '2.0',
+            '#appSecret' => '654321',
+        ];
+        $oauthResponse2 = $encryptorFactory->getEncryptor()->encrypt($credentials2);
+        $oauthClientStub2->expects($this->once())
+            ->method('getDetail')
+            ->with('keboola.docker-demo', 'whatever')
+            ->will($this->returnValue($oauthResponse2));
+        $oauthClientStub3 = self::getMockBuilder(Credentials::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $credentials3 = [
+            'id' => 'test-credentials-45',
+            'authorizedFor' => '',
+            'creator' => [
+                'id' => '3800',
+                'description' => 'ondrej.hlavacek@keboola.com'
+            ],
+            'created' => '2016-02-09 09:47:16',
+            '#data' => '{"access_token":"abcd","token_type":"bearer","uid":"efgh"}',
+            'oauthVersion' => '2.0',
+            'appKey' => '123456',
+            '#appSecret' => '654321',
+        ];
+        $oauthResponse3 = $encryptorFactory->getEncryptor()->encrypt($credentials3);
+        $oauthClientStub3->expects($this->once())
+            ->method('getDetail')
+            ->with('keboola.docker-demo', 'whatever')
+            ->will($this->returnValue($oauthResponse3));
+
+        /** @var Credentials $oauthClientStub2 */
+        /** @var Credentials $oauthClientStub3 */
+
+        $config = ['oauth_api' => ['id' => 'whatever', 'version' => '3']];
+        $auth = new Authorization($oauthClientStub2, $oauthClientStub3, $encryptorFactory->getEncryptor(), 'keboola.docker-demo', false);
+        self::assertEquals(
```

Comment:
`assertEquals` neporovnává identitu objektů, ale jejich atributy, a podle testu mají `$credentials2` a `$credentials3` stejné hodnoty v atributech, takže ten test nechytí případnou chybu

https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertEquals

Ba ne, jsou jiný, ale na první pohled mě to netrklo, díval jsem se na `#appSecret` a `#data`. Ale jak je to black box, tak nevím, co `$client->getDetail()` vrací a jestli to fakt chytí změnu.

Reply:
tak jsem je zjinil